### PR TITLE
feat: manage selected area with keyboard

### PR DESCRIPTION
**Pull request by Michota**
# What's new?
- Create `tryExpandingTowardsDirection` utility function, that (as the name suggest) will try to expand (*or shrink!) selection in selected direction
    *if you want to shrink the selection, it will re-calculate new selection area.
- Add **shortcuts** for area selecting, and focused cell change.

#### Space

-  **Shift + Space**: Resizes the selection to include all columns in the current rows.
-  **Ctrl / Command + Space**: Resizes the selection to include all rows in the current columns.

#### Shift

-  **Shift + Up Arrow**: Resizes the selection upwards.
-  **Shift + Right Arrow**: Resizes the selection to the right.
-  **Shift + Down Arrow**: Resizes the selection downwards.
-  **Shift + Left Arrow**: Resizes the selection to the left.

#### CTRL / Command

-  **Ctrl / Command + Shift + Arrow Up**: Expand the selection to the cell in first row.
-  **Ctrl / Command + Shift + Arrow Down**: Expand the selection to the cell in last row.
-  **Ctrl / Command + Shift + Arrow Right**: Expand the selection to the cell in last column.
-  **Ctrl / Command + Shift + Arrow Left**: Expand the selection to the cell in first column.

## Focus

- **Ctrl / Command + Arrow Up**: Jump to the first row.
- **Ctrl / Command + Arrow Down**: Jump to the last row.
- **Ctrl / Command + Arrow Left**: Jump to the first column.
- **Ctrl / Command + Arrow Right**: Jump to the last column.
- **Ctrl / Command + Home**: Jump to the first row.
- **Ctrl / Command + End**: Jump to the last row.

# What has changed?
- Refactored `handleKeyDown.ts` for better redability maintability.

### CODE REVIEW

**Comment by Michota** [comment]
# Bug found: 
Changing focus with Ctrl + Arrows (jump), End/Home can cause issues
- focused cell may overlap selection
- span focused cell may be split in half.

**I will reopen the PR after applying the fixes**

**Comment by Michota** [comment]
Focus-jump feature (related to shortcuts, e.g. `CTRL + ArrowUp`) was refactored so it will work similarly to other spreadsheet software/apps like Excel, or Google Spreadsheets.

**Review by DLowHP — APPROVED** [review]
I added a few minor changes, mainly related to comments. I've also tested this and it looks to be working right.

**Review by DLowHP — APPROVED** [review]

File: src/utils/resizeTowardsDirection.ts
Comment (MichaelMatejko) [inline src/utils/resizeTowardsDirection.ts:1]: resizeInDirection

**Review by MichaelMatejko — APPROVED** [review]

**Pull request merged**